### ISSUE TITLE
Grid block doesn't adopt to Zoom out view

### ISSUE THREAD

**Issue by bph** [WordPress/gutenberg#65974]
### Description

When using a pattern with a grid block of images on a page, and then switch to zoom out view, the grid expands outside the Zoom out space and also breaks it from a subsequent section on the page. 

### Step-by-step reproduction instructions

1. Use Twenty-Twenty Five theme
2. Create a new page
3. Add the Galleries > Instagram grid block 
4. Save the page
5. Use the toogle button to Zoom out. 

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/9b01010c-d41d-49a3-82d5-6b1f277e6cf8

### Environment info

- WordPress 6.7-beta2-59195
- Twenty-Twenty-Five
- Google Chrome Version 129.0.6668.90
- MacOS Sonoma 14.5

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by bph** [comment]
The only Grid setting that works well with the zoom out mode is when I set it to **Manuell**. 

![Image](https://github.com/user-attachments/assets/2385856f-9b0e-4444-acca-e74d0613bc3e)

**Comment by bph** [comment]
A longer video of me testing different grid block settings in available on **Cloudup**
https://cloudup.com/cYSG-xVQzs4

**Comment by MaggieCabrera** [comment]
Let's look into this when https://github.com/WordPress/gutenberg/pull/65814 gets merged, I think it improves it a lot

**Comment by bph** [comment]
I tested #65814  via Playground and it is definitely an improvement. The only quirk seems to remain are the grid lines still expanding the zoom space but the images in the grid stay inside the white background. Tested both wide align and no align. See screenshot
![Image](https://github.com/user-attachments/assets/63dd102b-0d79-4ce6-a931-58b04815a9b0)

**Comment by aaronrobertshaw** [comment]
I was about to report that I couldn't replicate this after https://github.com/WordPress/gutenberg/pull/65998 was merged but before posting I did spot a flash of the grid visualizer extending beyond what it should in the video below around the 30 second mark.

https://github.com/user-attachments/assets/bb8ff676-73e4-434a-a61c-1031f4317e0e

The edge case in that video seemed to be when I was randomly clicking and double clicking around on the grid pattern.

Following that further I think I found a way to reliably replicate the issue.

1. Add instagram grid pattern
2. Drag select an image within the block
3. Engage zoom out view
4. Click into the grid pattern
5. Note the grid visualizer is present

https://github.com/user-attachments/assets/cf9338d6-4911-4c6b-ab1a-ea9e3e5b343c

I've run out of time to dig deeper but maybe there's something we could do around tightening up when the grid block's visualizer is shown?
